packages/cli-repl/test/e2e-tls.spec.ts

```diff
@@ -156,14 +183,82 @@ describe('e2e TLS', () => {
+        if (process.platform !== 'linux') {
+          return this.skip();
+        }
+        await fs.mkdir(path.join(tmpdir.path, 'certs'), { recursive: true });
+        await fs.copyFile(CA_CERT, path.join(tmpdir.path, 'certs', 'somefilename.crt'));
+        const shell = TestShell.start({
+          args: [
+            `${await server.connectionString()}?serverSelectionTimeoutMS=1500`,
+            '--tls', '--tlsUseSystemCA'
+          ],
+          env: {
+            ...env,
+            SSL_CERT_DIR: path.join(tmpdir.path, 'certs') + ':/nonexistent/other/path'
+          }
+        });
+
+        const prompt = await shell.waitForPromptOrExit();
+        expect(prompt.state).to.equal('prompt');
+
+        const logPath = path.join(logBasePath, `${shell.logId}_log`);
+        const logContents = await readReplLogfile(logPath);
+        expect(logContents.find(line => line.id === 1_000_000_049).attr.asyncFallbackError)
+          .to.equal(null); // Ensure that system CA loading happened asynchronously.
+      });
+
+      it('works with system CA on Linux with SSL_CERT_FILE', async function() {
+        if (process.platform !== 'linux') {
+          return this.skip();
+        }
+        await fs.mkdir(path.join(tmpdir.path, 'certs'), { recursive: true });
+        await fs.copyFile(CA_CERT, path.join(tmpdir.path, 'certs', 'somefilename.crt'));
+        const shell = TestShell.start({
+          args: [
+            `${await server.connectionString()}?serverSelectionTimeoutMS=1500`,
+            '--tls', '--tlsUseSystemCA'
+          ],
+          env: {
+            ...env,
+            SSL_CERT_FILE: path.join(tmpdir.path, 'certs', 'somefilename.crt')
+          }
+        });
+
+        const prompt = await shell.waitForPromptOrExit();
+        expect(prompt.state).to.equal('prompt');
+
+        const logPath = path.join(logBasePath, `${shell.logId}_log`);
+        const logContents = await readReplLogfile(logPath);
+        expect(logContents.find(line => line.id === 1_000_000_049).attr.asyncFallbackError)
+          .to.equal(null); // Ensure that system CA loading happened asynchronously.
+      });
+
+      it('fails on macOS/Windows with system CA', async function() {
+        // No good way to programmatically add certs to the system CA from our tests.
+        if (process.platform !== 'darwin' && process.platform !== 'win32') {
+          return this.skip();
+        }
+        const shell = TestShell.start({
+          args: [
+            `${await server.connectionString()}?serverSelectionTimeoutMS=1500`,
+            '--tls', '--tlsUseSystemCA'
+          ],
+          env
+        });
+
+        const prompt = await shell.waitForPromptOrExit();
+        expect(prompt.state).to.equal('exit');
+
+        const logPath = path.join(logBasePath, `${shell.logId}_log`);
+        const logContents = await readReplLogfile(logPath);
+        expect(logContents.find(line => line.id === 1_000_000_049)).to.exist;
```

Comment:
If this test checks that the loading from system CA should fail on these systems, wouldn't this log line be missing from the log file? IIUC `1_000_000_049` shows in the logs when we managed to read from system CA